packages/next/src/build/templates/app-page.ts

```diff
@@ -1213,19 +1209,22 @@ export async function handler(
         res.setHeader(NEXT_DID_POSTPONE_HEADER, '1')
       }
 
-      // we don't go through this block when preview mode is true
-      // as preview mode is a dynamic request (bypasses cache) and doesn't
-      // generate both HTML and payloads in the same request so continue to just
-      // return the generated payload
-      if (isRSCRequest && !isDraftMode) {
+      // NOTE: We previously excluded requests that were in draft mode because
+      // the logic here used to also serve pages router data requests. Now that
+      // this only serves app router data requests, the error branches within
+      // this block are appropriately gated. The only error conditions that are
+      // possible are if the response truly wasn't an RSC response.
+      if (isRSCRequest) {
         // If this is a dynamic RSC request, then stream the response.
         if (typeof cachedData.rscData === 'undefined') {
           // If the response is not an RSC response, then we can't serve it.
           if (cachedData.html.contentType !== RSC_CONTENT_TYPE_HEADER) {
+            // TODO: (wyattjoh) this may an unreachable branch, see if we can remove this
             if (nextConfig.experimental.clientParamParsing) {
-              // If client param parsing is enabled, then we can return a 404.
-              // This was likely an old prefetch request.
-              res.statusCode = 404
+              // If client param parsing is enabled, then we can return a 422.
+              // This was likely an old prefetch request which we don't return
+              // anymore with the client param parsing enabled.
+              res.statusCode = 422
```

Comment:
nothing relies on a 404 here? feels like another change that we could have put in a different PR